app/controllers/users.js

```diff
@@ -1,9 +1,30 @@
+const jwt = require('jsonwebtoken');
+
+const config = require('../../config');
+const { expiresIn } = require('../constants');
+
 const { signUp } = require('../mappers/users');
-const { saveUser } = require('../services/users');
+const { saveUser, verifyCredentials } = require('../services/users');
 
 exports.createUser = (req, res) => {
   const dataUser = signUp(req.body);
   saveUser(dataUser)
     .then(response => res.status(200).send(response))
     .catch(err => res.status(400).send(err));
 };
+
+exports.signInUser = (req, res) => {
+  verifyCredentials(req.body.email, req.body.password)
+    .then(response => {
+      const Token = jwt.sign(
+        {
+          name: response.user.name,
+          last_name: response.user.last_name
+        },
+        config.common.seedToken,
+        { expiresIn }
+      );
+      res.status(200).send({ token: Token });
+    })
+    .catch(err => res.status(400).send(err));
```

Comment:
for this case I would expect more personalized errors, making use of other status for example for this case a 401 seems to me to be the ideal

Reply:
But the state 401 is when the user tries to access a one module without authentication, in this case he is entering values for get a token. I believe that state 400 is more appropriate

Reply:
you're right leave it like this